pyomo/contrib/pynumero/src/CMakeLists.txt

```diff
@@ -0,0 +1,167 @@
+  PATH_SUFFIXES asl
+)
+FIND_LIBRARY(ASL_LIBRARY NAMES coinasl asl
+  HINTS "${CMAKE_INSTALL_PREFIX}/lib"
+        "${IPOPT_DIR}/lib"
+        "${AMPLMP_DIR}/lib"
+        "${PC_COINASL_LIBDIR}"
+        "${PC_COINASL_LIBRARY_DIRS}"
+)
+FIND_LIBRARY(HSL_LIBRARY NAMES coinhsl libcoinhsl
+  HINTS "${CMAKE_INSTALL_PREFIX}/lib"
+        "${IPOPT_DIR}/lib"
+        "${PC_COINHSL_LIBDIR}"
+        "${PC_COINHSL_LIBRARY_DIRS}"
+)
+
+# If BUILD_AMPLMP_IF_NEEDED is set and we couldn't find / weren't
+# pointed to an ASL build, then we will forcibly enable the AMPLMP build
+# to provide the ASL.
+IF( BUILD_AMPLMP_IF_NEEDED AND (NOT ASL_LIBRARY OR NOT ASL_INCLUDE_DIR) )
+    set_property(CACHE BUILD_AMPLMP PROPERTY VALUE ON)
+ENDIF()
+
+IF( BUILD_AMPLMP )
+  get_filename_component(ABS_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" ABSOLUTE)
+  ExternalProject_Add(amplmp
+    GIT_TAG ${AMPLMP_TAG}
+    GIT_REPOSITORY https://github.com/ampl/mp.git
+    # We don't need *any* submodules, but leaving it as an empty string
+    # doesn't disable it as suggested by the documentation.  A
+    # "workaround" from the web is to specify an existing directory that
+    # is *not* a submodule
+    GIT_SUBMODULES test
+    CMAKE_CACHE_ARGS -DCMAKE_INSTALL_PREFIX:STRING=${ABS_INSTALL_PREFIX}
+    UPDATE_DISCONNECTED TRUE
+    # 3.1.0 needs to be patched to compile with recent compilers,
+    # notably ubuntu 18.04.  The patch applies a backport of fmtlib/fmt
+    # abbefd7; see https://github.com/fmtlib/fmt/issues/398
+    PATCH_COMMAND git apply
+       ${CMAKE_CURRENT_SOURCE_DIR}/amplmp-${AMPLMP_TAG}.patch
+    )
+  # Update the ASL paths (if necessary).  Since these do not (yet)
+  # exist, we need to bypass find_path / find_library and explicitly set
+  # the directories that this build will create.  However, we will only
+  # do this if the paths have not already been set (so users can always
+  # override what we do here)
+  IF(NOT ASL_INCLUDE_DIR OR NOT ASL_LIBRARY)
+    set_property(CACHE ASL_INCLUDE_DIR PROPERTY VALUE
+      "${ABS_INSTALL_PREFIX}/include/asl")
+    IF( WIN32 )
+      set_property(CACHE ASL_LIBRARY PROPERTY VALUE
+        "${ABS_INSTALL_PREFIX}/lib/asl.lib")
+    ELSE()
+      set_property(CACHE ASL_LIBRARY PROPERTY VALUE
+        "${ABS_INSTALL_PREFIX}/lib/libasl.a")
+    ENDIF()
+  ENDIF()
+ENDIF()
+
+set(PYNUMERO_ASL_SOURCES
+  "AmplInterface.cpp"
+  "AmplInterface.hpp"
+  "AssertUtils.hpp"
+)
+
+IF( BUILD_ASL )
+  ADD_LIBRARY( pynumero_ASL SHARED ${PYNUMERO_ASL_SOURCES} )
+  TARGET_LINK_LIBRARIES( pynumero_ASL PUBLIC ${ASL_LIBRARY} )
+  if ( DL_LIBRARY )
+    TARGET_LINK_LIBRARIES( pynumero_ASL PUBLIC ${DL_LIBRARY} )
+  ENDIF()
+  TARGET_INCLUDE_DIRECTORIES( pynumero_ASL
+    PUBLIC ${ASL_INCLUDE_DIR}
+    INTERFACE . )
+  TARGET_COMPILE_DEFINITIONS( pynumero_ASL PRIVATE BUILDING_PYNUMERO_ASL )
+  SET_TARGET_PROPERTIES( pynumero_ASL PROPERTIES ENABLE_EXPORTS 1 )
+  INSTALL(TARGETS pynumero_ASL LIBRARY DESTINATION lib )
```

Comment:
I had to add ```RUNTIME DESTINATION lib```.  That prevented an error with MinGW64 running in MSYS2 on Windows.  I didn't need the install target though, so I can say for sure whether it is correct.

Reply:
@eslickj, to confirm, the following worked on all platforms:
```cmake
INSTALL(TARGETS pynumero_ASL
    LIBRARY DESTINATION lib
    RUNTIME DESTINATION lib
)
```